aws/aws.go

```diff
@@ -188,9 +188,10 @@ func V2ConfigFromURLParams(ctx context.Context, q url.Values) (awsv2.Config, err
 			customResolver := awsv2.EndpointResolverWithOptionsFunc(
 				func(service, region string, options ...interface{}) (awsv2.Endpoint, error) {
 					return awsv2.Endpoint{
-						PartitionID:   "aws",
-						URL:           value,
-						SigningRegion: region,
+						PartitionID:       "aws",
+						URL:               value,
+						SigningRegion:     region,
+						HostnameImmutable: true,
```

Comment:
The documentation says "Setting an endpoint as immutable may prevent some service client features from functioning correctly, and could result in undefined behavior. Caution should be taken when defining an endpoint as immutable."

https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/endpoints/#immutable-endpoints

Should we make this an option instead of always setting it to true? Or is it reasonable to expect it to always be true when someone is explicitly setting an endpoint?

Reply:
I think it should be an option, preventing unexpected behavior for others users

Reply:
yeah, I was thinking it too, gonna work on it later today

Reply:
done